alacritty_terminal/src/term/mod.rs

```diff
@@ -2429,6 +2733,277 @@ mod tests {
+        let right = term.columns() - 1;
+        let mut selection =
+            Selection::new(SelectionType::Block, Point::new(Line(line), Column(3)), Side::Left);
+        selection.update(Point::new(Line(line), Column(5)), Side::Left);
+        let selection_range = selection.to_range(&term);
+
+        let mut damaged_lines = match term.damage(selection_range) {
+            TermDamage::Full => panic!("Expected partial damage, however got Full"),
+            TermDamage::Partial(damaged_lines) => damaged_lines,
+        };
+        let line = line as usize;
+        // Skip information about cursor on zero line, since we just check for selection.
+        damaged_lines.next();
+        assert_eq!(damaged_lines.next(), Some(LineDamageBounds { line, left, right }));
+        assert_eq!(damaged_lines.next(), None);
+        term.reset_damage();
+
+        // Check that existing selection gets damaged when it gets removed.
+        let mut damaged_lines = match term.damage(None) {
+            TermDamage::Full => panic!("Expected partial damage, however got Full"),
+            TermDamage::Partial(damaged_lines) => damaged_lines,
+        };
+        // Skip information about cursor on zero line.
+        damaged_lines.next();
+        assert_eq!(damaged_lines.next(), Some(LineDamageBounds { line, left, right }));
+        assert_eq!(damaged_lines.next(), None);
+        term.reset_damage();
+
+        // Check that `Vi` cursor in vi mode is being damaged.
+        term.toggle_vi_mode();
+        // Put Vi cursor to a different location than normal cursor.
+        term.vi_goto_point(Point::new(Line(5), Column(5)));
+        // Reset damage, so the damage information from `vi_goto_point` won't affect test.
+        term.reset_damage();
+        let vi_cursor_point = term.vi_mode_cursor.point;
+        let line = vi_cursor_point.line.0 as usize;
+        let left = vi_cursor_point.column.0 as usize;
+        let right = left;
+        let mut damaged_lines = match term.damage(None) {
+            TermDamage::Full => panic!("Expected partial damage, however got Full"),
+            TermDamage::Partial(damaged_lines) => damaged_lines,
+        };
+        // Skip information about cursor on zero line.
+        damaged_lines.next();
+        assert_eq!(damaged_lines.next(), Some(LineDamageBounds { line, left, right }));
+        assert_eq!(damaged_lines.next(), None);
+    }
+
+    #[test]
+    fn damage_cursor_movements() {
+        let size = SizeInfo::new(10.0, 10.0, 1.0, 1.0, 0.0, 0.0, false);
+        let mut term = Term::new(&Config::default(), size, ());
+        let num_cols = term.columns();
+        // Reset terminal for partial damage tests since it's initialized as fully damaged.
+        term.reset_damage();
+
+        term.goto(Line(1), Column(1));
+
+        // NOTE We want to test what damage information is being produced by each function
+        // individually and the entire damage that we could get from `[Term::damage]`.
```

Comment:
What is the purpose of this comment? How does it relate to the code it is annotating?

You're telling me you want to do something, but I see no relation between that and what is commented here? You talk about damage being produced by each function but call no function that produces damage. You talk about damage you get from `Term::damage` but read the damage from `term.damage` directly. It doesn't make sense?